partiql-tests-data/eval/primitives/path.ion

```diff
@@ -512,3 +512,296 @@ pathUnpivotMissing::[
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:"this is unambiguous_lookup_field"
+    }
+  },
+  {
+    name:"subscript with literal string in lowercase",
+    statement:"test_struct['ambiguous_lookup_field']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:"this is ambiguous_lookup_field (lowercase)"
+    }
+  },
+  {
+    name:"subscript with literal string in uppercase",
+    statement:"test_struct['AMBIGUOUS_LOOKUP_FIELD']",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:"this is AMBIGUOUS_LOOKUP_FIELD (uppercase)"
+    }
+  },
+  {
+    name:"subscript with non-existent variable in lowercase",
+    statement:"test_struct[unambiguous_field]",
+    assert:[
+      {
+        evalMode:EvalModeError,
+        result:EvaluationFail
+      },
+      {
+        result:EvaluationSuccess,
+        evalMode:EvalModeCoerce,
+        output:$missing::null
+      }
+    ]
+  },
+  {
+    name:"subscript with non-existent variable in uppercase",
+    statement:"test_struct[TEST_FIELD]",
+    assert:[
+      {
+        evalMode:EvalModeError,
+        result:EvaluationFail
+      },
+      {
+        result:EvaluationSuccess,
+        evalMode:EvalModeCoerce,
+        output:$missing::null
+      }
+    ]
+  },
+  {
+    name:"path expression with table alias in uppercase",
+    statement:"SELECT AN_ALIAS.f FROM table1 as an_alias",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+        {
+          f:"this is from table1"
+        }
+      ]
+    }
+  },
+  {
+    name:"path expression with table alias in lowercase",
+    statement:"SELECT an_alias.f FROM table1 as AN_ALIAS",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+        {
+          f:"this is from table1"
+        }
+      ]
+    }
+  },
+  {
+    name:"path expression with ambiguous table alias (lowercase)",
+    statement:"SELECT \"an_alias\".f FROM table1 AS an_alias, table2 AS AN_ALIAS",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+        {
+          f:"this is from table1"
+        }
+      ]
+    }
+  },
+  {
+    name:"path expression with ambiguous table alias (uppercase)",
+    statement:"SELECT \"AN_ALIAS\".f FROM table1 AS an_alias, table2 AS AN_ALIAS",
+    assert:{
+      result:EvaluationSuccess,
+      evalMode:[
+        EvalModeCoerce,
+        EvalModeError
+      ],
+      output:$bag::[
+        {
+          f:"this is from table2"
+        }
+      ]
+    }
+  }
+]
+
+'repeated-field-on-struct'::[
```

Comment:
The COERCE/permissive mode behavior is implementation-dependent as per section 4 of spec. Hence why only the ERROR/strict mode behavior tests are defined.